rasa/core/policies/ted_policy.py

```diff
@@ -970,78 +979,90 @@ def _load_model_utilities(cls, model_path: Path) -> Dict[Text, Any]:
     @classmethod
     def load(
         cls,
-        path: Union[Text, Path],
-        should_finetune: bool = False,
-        epoch_override: int = defaults[EPOCHS],
-        **kwargs: Any,
-    ) -> "TEDPolicy":
-        """Loads a policy from the storage.
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> TEDPolicy2:
+        """Loads a policy from the storage (see parent class for full docstring)."""
+        try:
```

Comment:
how do we feel about indentation levels?

Reply:
I think you're absolutely right 👍🏻 Due to the current state of our featurizers I think it makes sense to keep it like this _for now_ though because
- We currently only have one `Resource` per node (related to your question). As soon as featurizers would become their own graph nodes they'd also get their own `Resource` and we can then adapt them as required. Currently they'd end up writing to the same directory anyway.
- For the moment there is no need to expose this information to the featurizers. I agree that we should do this mid/long term but as long as they are embedded within policies we in my opinion don't need to expose the outer structure to them which also
- keeps the diff and changes smaller. In my opinion we should reach a stable system with the minimum of required changes first and then decide which bells and whistles we add. 

Does that make sense? What do you think?

Reply:
Yep makes sense. You're right, it's internals of TED and so TED has to make sure it uses that correctly and we never use it in such a dynamic way that adding automatic checks elsewhere makes much sense. Not sure that featurizers will ever become graph components (With the disentangling+sequence-generators/mini-batch work, it should even be more lightweight and more tightly coupled with TED).